File: app/src/main/java/com/cn29/aac/ui/viewpager/vm/PagerAgentViewModel.java
package com.cn29.aac.ui.viewpager.vm;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

/**
 * Created by charlesng0209 on 19/6/2017.
 */

public class PagerAgentViewModel extends ViewModel {
    private MutableLiveData<String> messageContainerA;
    private MutableLiveData<String> messageContainerB;

    public void init()
    {
        messageContainerA = new MutableLiveData<>();
        messageContainerA.setValue("Default Message");
        messageContainerB = new MutableLiveData<>();
        messageContainerB.setValue("Default Message");
    }

    public void sendMessageToB(String msg)
    {
        messageContainerB.setValue(msg);
    }
    public void sendMessageToA(String msg)
    {
        messageContainerA.setValue(msg);

    }
    public LiveData<String> getMessageContainerA() {
        return messageContainerA;
    }

    public LiveData<String> getMessageContainerB() {
        return messageContainerB;
    }
}

File: app/src/main/java/com/cn29/aac/ui/viewpager/vm/PagerAgentViewModel.kt
package com.cn29.aac.ui.viewpager.vm

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

/**
 * Created by charlesng0209 on 19/6/2017.
 */
class PagerAgentViewModel : ViewModel() {
    private var messageContainerA: MutableLiveData<String>? = null
    private var messageContainerB: MutableLiveData<String>? = null
    fun init() {
        messageContainerA = MutableLiveData()
        messageContainerA?.value = "Default Message"
        messageContainerB = MutableLiveData()
        messageContainerB?.value = "Default Message"
    }

    fun sendMessageToB(msg: String) {
        messageContainerB?.value = msg
    }

    fun sendMessageToA(msg: String) {
        messageContainerA?.value = msg
    }

    fun getMessageContainerA(): LiveData<String>? {
        return messageContainerA
    }

    fun getMessageContainerB(): LiveData<String>? {
        return messageContainerB
    }
}